src/relay/transforms/gradient.cc

```diff
@@ -438,12 +449,17 @@ Expr BPEmpty() {
 
 struct ReverseAD : ExprMutator {
   using ADVarMap = std::unordered_map<Var, Var, ObjectPtrHash, ObjectPtrEqual>;
-
+  using ADGVarMap = std::unordered_map<GlobalVar, GlobalVar, ObjectPtrHash, ObjectPtrEqual>;
+  Optional<IRModule> mod;
```

Comment:
Is this really optional?

Reply:
yes - sometime it is call on naked expr. i could refactor to require it to always be here, if this is what you want

Reply:
Can you just add a note about this to your on-going doc on things we might want to refactor?

Reply:
you mean the list from ~2 years ago? that is very out of dated. I think we can just grep -r for TODO(@M.K.) in the source.